fighthealthinsurance/ml/ml_models.py

```diff
@@ -426,10 +522,29 @@ async def _checked_infer(
                 pubmed_context=pubmed_context,
                 temperature=temperature,
                 ml_citations_context=ml_citations_context,
-            )
-        # Ok just an empty list, we failed
-        if self.bad_result(result, infer_type):
-            return []
+            )        
+            # Ok just an empty list, we failed
+            if self.bad_result(result, infer_type):
+                return []
+        
+        logger.debug(f"Checking if professional")
+       
+        # If professional_to_finish then check if the result is a professional response | One retry | Logs inside is_professional_tone
+        if prof_pov:
+            if not self.is_professional_tone(result):
+                result = await self._infer_no_context(
+                    prompt=prompt,
+                    patient_context=patient_context,
+                    plan_context=plan_context,
+                    system_prompts=[system_prompt],
+                    pubmed_context=pubmed_context,
+                    temperature=temperature,
+                    ml_citations_context=ml_citations_context,
+                )        
+                if self.bad_result(result, infer_type):
+                    return []
+                if not self.is_professional_tone(result):
+                    return []
```

Comment:
Here I'd rather us default to returning something if after two attempts it's still patient voice.